src/api/ledgerManager.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import {Ledger} from "../core/ledger/ledger";
+import type {LedgerLog} from "../core/ledger/ledger";
+import diffBy from "lodash.differenceby";
+
+/**
+ * Generic adaptor for persisting a JSON log to some storage backend
+ * (e.g. GitHub, local filesystem, a database, etc)
+ */
+export interface LedgerStorage {
+  read(): Promise<Ledger>;
+  write(ledger: Ledger): Promise<void>;
+}
+
+type LedgerManagerConfig = {|
+  +storage: LedgerStorage,
+  +initLogs?: LedgerLog,
```

Comment:
why is this argument optional, and why is it the log and not just a Ledger? smells funny to me

Reply:
The reason it's optional is because the Ledger manager can just load its state from the remote storage. The main reason having it here was more-so for testing purposes, but there could be more practical reasons to have it here (e.g. instantiating ledger instance from a users local ledger in case its not the same as the ledger persisted in GH)

The reason it takes the LedgerLog instead of the Ledger is because we want the LedgerManager to encapsulate its own instance of the ledger. We need to ensure that the user always goes through the LM when making changes instead of using their own external instance of a ledger since the reference to that instance would get thrown away by the LedgerManager when it does a re-load and there's new remote changes to sync. You can see in this line here that the internal ledger is reassigned to a new instance: https://github.com/sourcecred/sourcecred/pull/2480/files#diff-b1a73d3817f14a44e23c07abab8e24108919774aecf52e6cb8fa220668b7aaa9R96

Right now we still allow the user to make any method calls on the internal ledger instance, but in the future we could add specific API's that restrict ledger interaction to a subset of these, or add additional checks / verifications before forwarding events/actions to the underlying ledger.